modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/CacheWorker.kt

```diff
@@ -95,8 +103,38 @@ class CacheWorker @AssistedInject constructor(
                 .setConstraints(constraints)
                 .setInputData(inputData).build()
 
+            observeWorkerInfo(context, cacheWorkRequest, episodeUuid, onCachingComplete)
+
             // Enqueue new caching work
             WorkManager.getInstance(context).enqueue(cacheWorkRequest)
         }
+
+        private fun observeWorkerInfo(
+            context: Context,
+            cacheWorkRequest: OneTimeWorkRequest,
+            episodeUuid: String,
+            onCachingComplete: (String) -> Unit,
+        ) {
+            ProcessLifecycleOwner.get().lifecycleScope.launch {
+                WorkManager.getInstance(context)
+                    .getWorkInfoByIdFlow(cacheWorkRequest.id).collectLatest { workInfo ->
+                        if (workInfo?.state == WorkInfo.State.SUCCEEDED &&
+                            workInfo.outputData.getString(EPISODE_UUID_KEY) == episodeUuid
+                        ) {
+                            /* For fully cached media, ExoPlayer's buffer position might not reach the end,
+                            even when the entire media is cached. This is due to ExoPlayer's LoadControl
+                            mechanism, which might not request the entire media to be buffered even when it's
+                            fully cached.
+
+                            This callback is triggered when the media is fully cached and is used to
+                            manually update the buffer position to the end of the media, ensuring more accurate
+                            buffer status tracking.
+
+                            See https://rb.gy/333bzt for more details. */
```

Comment:
Would you mind using the full URL instead of the shortened one? If `rb.gy` goes out of business we won't have access to the details.

Reply:
https://github.com/Automattic/pocket-casts-android/pull/2987/commits/f25c7abe38de9e7cf69fe4b2d0428b342ef6969e